inventory_reporting/reports/inventory_report.py

```diff
@@ -227,12 +227,20 @@ def generate_storable_report(self, workbook, wizard):
                 )
 
     def generate_consumable_report(self, workbook, wizard):
-        base_consu_domain = [
-            ("stock_move_id.date", ">=", wizard.date_start),
-            ("stock_move_id.date", "<=", wizard.date_end),
-            ("product_id.active", "=", True),
-            ("product_id.detailed_type", "!=", "product"),
-        ]
+        query = """
+            SELECT svl.id FROM stock_valuation_layer svl
+            JOIN stock_move move ON svl.stock_move_id = move.id
+            JOIN stock_picking pick ON move.picking_id = pick.id
+            JOIN product_product prod ON svl.product_id = prod.id
+            JOIN product_template tmpl ON prod.product_tmpl_id = tmpl.id
+            WHERE COALESCE(pick.accounting_date, move.date) BETWEEN %s AND %s
+            AND prod.active = TRUE
+            AND tmpl.detailed_type != 'product'
+        """
+        self.env.cr.execute(query, (wizard.date_start, wizard.date_end))
+        svl_ids = [row[0] for row in self.env.cr.fetchall()]
+
+        base_consu_domain = [("id", "in", svl_ids)]
```

Comment:
I thought I posted a rather lengthy comment on this in the last review but I guess I forgot to post it.  I am not comfortable with this change and we should consider adding the `accounting_date` field in stock.move (in stock_picking_accounting_date, and inherit and adjust the relevant method in stock_picking_consumable_effective_date) and use this field in `base_consu_domain` (consider consolidating the common parts of `base_consu_domain` and `base_product_domain` into one as appropriate.  Please identify the existing logic that needs to be adjusted and follow up.

Reply:
Hi @AungKoKoLin1997 , can you please adjust this line as well in for the consumable report.
I think this field should reflect the date that set in the consumable effective date field or the one from the effective date if consumable effective date is none.

Reply:
My mistake. I didn't notice we have https://github.com/qrtl/axls-custom/tree/16.0/stock_move_accounting_date. I will adjust depends on this module.